internal/fetchclient/fetchclient.go

```diff
@@ -321,17 +247,45 @@ func (c *Client) fetchMavenPage(ctx context.Context, targetURL *url.URL, group s
 	}
 	defer response.Body.Close()
 	if response.StatusCode != http.StatusOK {
-		if errorBody, err := io.ReadAll(response.Body); err != nil {
-			log.Println("error:", string(errorBody))
-		}
 		return nil, fmt.Errorf("received status code %d retrieving %q", response.StatusCode, request.URL.String())
 	}
-
-	var data mavenResponse
-	if err := json.NewDecoder(response.Body).Decode(&data); err != nil {
+	var metadata struct {
+		GroupID    string `xml:"groupId"`
+		ArtifactID string `xml:"artifactId"`
+		Versioning struct {
+			Latest      string   `xml:"latest"`
+			Release     string   `xml:"release"`
+			Versions    []string `xml:"versions>version"`
+			LastUpdated string   `xml:"lastUpdated"`
+		} `xml:"versioning"`
+	}
+	if err := xml.NewDecoder(response.Body).Decode(&metadata); err != nil {
 		return nil, err
 	}
-	return &data, nil
+	results := &mavenResults{
+		latestVersion:       "",
+		latestPatchVersions: make(map[string]string),
+	}
+	for _, version := range metadata.Versioning.Versions {
+		if !strings.HasPrefix(version, "v") {
+			version = "v" + version
+		}
+		if !semver.IsValid(version) || semver.Prerelease(version) != "" {
+			continue
+		}
+		version = semver.Canonical(version)
+		if results.latestVersion == "" || semver.Compare(results.latestVersion, version) < 0 {
+			results.latestVersion = version
+		}
+		release := semver.MajorMinor(version)
```

Comment:
Can you remind me why we're doing this extra bit of logic for maven in addition to the semver.Compare above?

Reply:
The hope was that for older versions of plugins (that have new patch releases available for security issues) that we'd also discover those as new plugin versions in addition to the latest version changing.

Reply:
At the end we only surface a single version, so if we have [v1.2.3, v1.2.4, v1.3.0] then v1.3.0 will get returned (even if v1.2.4 was also included). The initial sort should ensure the latest semver gets returned?

I'm also not convinced we need this extra logic, because we only care about the latest semver. I.e., if you're on v1.2.3 and v1.3.0 is available, why do you need v1.2.4? If this is an issue for end-users we could always backfill a version, but I suspect for 99% of users the latest should suffice?

Reply:
I think the thing that confuses me is that we never use `latestPatchVersions`, so every time I stumble on the maven fetch this logic trips me up 😅 . Maybe you can add a comment as to why this is different to all the other fetchers?